Bump tesla-fleet-api to 1.4.2

Bre77 opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Bump Tesla Fleet API to 1.4.0 in order to add strict typing
https://github.com/Teslemetry/python-tesla-fleet-api/compare/v1.3.2...v1.4.2

This did require a number of other changes to make usage type safe too.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [ ] I understand the code I am submitting and can explain how it works.
- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.
- [ ] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Bre77 committed:
Bump Tesla Fleet API to 1.4.0

Bre77 committed:
Bump Tesla Fleet API to 1.4.0

Bre77 committed:
Fix region type

Bre77 committed:
make api usage consistant

Bre77 committed:
Dont use or 0

joostlek commented on homeassistant/components/tesla_fleet/__init__.py:
```diff
@@ -173,14 +179,16 @@ async def _get_access_token() -> str:
                 continue
 
-            api = tesla.energySites.create(site_id)
+            api_energy: EnergySite = tesla.energySites.create(site_id)
```
Not sure why we need to retype this, I'd expect mypy to infer the type here

Bre77 replied:
Removed.

joostlek commented on homeassistant/components/tesla_fleet/number.py:
```diff
@@ -64,7 +64,7 @@ class TeslaFleetNumberVehicleEntityDescription(NumberEntityDescription):
         min_key="charge_state_charge_limit_soc_min",
         max_key="charge_state_charge_limit_soc_max",
-        func=lambda api, value: api.set_charge_limit(value),
+        func=lambda api, value: api.set_charge_limit(int(value)),
```
You only pass on `int` here. Why not change the callable signature to be `Callable[[..], int]` instead? This way you don't have to explicitly do `int` everywhere (unless you expect something in the future I don't)

joostlek replied:
the lambda can guarnatee it by just making the lambda get the value as an int and move the `int` to the number entity

Bre77 replied:
Changed the description to require an int.

Bre77 committed:
Remove extra type

Bre77 committed:
Force int at the description level

Bre77 committed:
Add tests to cover tesla_fleet and teslemetry changes

joostlek reviewed (changes requested):
and CI is failing

claude committed:
Simplify region validation to use if statement

Replace try/assert/except pattern with a simple if statement
for region validation, as suggested in review feedback.

claude committed:
Update test comment to reflect region validation change

Bre77 committed:
Fix migration call

Bre77 committed:
Bump to 1.4.2

Bre77 committed:
Updating missing manifests

Bre77 committed:
Done reuse api
